### DISCUSSION TITLE
Error occurred when going backwards (getBoundingClientRect)

### DISCUSSION THREAD

**Discussion by devilofdev** [jeffreylanters/react-unity-webgl#434] [Help Wanted!]
Hi. i'm Stephen. Thanks to you, I am working hard to develop the service. :)

a getBoundingClientRect error occurs if you go back from the page where the Unity instance is activated.

<img width="441" alt="Screen Shot 2022-08-26 at 9 38 33 AM" src="https://user-images.githubusercontent.com/46062585/186792420-a5a763ff-eac0-4d36-9718-87b387c47879.png">

So before running the Backward event, I'm going to unload it.
Sometimes it works, but sometimes it doesn't work when you enter a page that brings up another scene.

below is my code.
```ts
 const handleUnload = useCallback(async () => {
    await unload();
  }, [unload]);

  useEffect(() => {
    router.beforePopState(({ as }) => {
      if (as !== router.asPath) {
        handleUnload();
        return true;
      } else {
        return true;
      }
    });

    return () => {
      router.beforePopState(() => true);
    };
  }, [handleUnload, router]);
```

can i get some help? n_n

**Comment by devilofdev** [comment]
i modify my code, after unity instantiate is completed, backward event is well done.
but.. when enter page it doesn't loaded, occur TypeError: Cannot read properties of null (reading 'addEventListener')

my modify code
```ts
const handleUnload = useCallback(async () => {
    await unload();
    router.push("/space");
  }, [router, unload]);

  useEffect(() => {
    router.beforePopState(({ as }) => {
      if (as !== router.asPath) {
        if (isLoaded) {
          handleUnload();
          return false;
        } else {
          return true;
        }
      } else {
        return true;
      }
    });

    return () => {
      router.beforePopState(() => true);
    };
  }, [handleUnload, router, isLoaded, removeEventListener]);
```

**Comment by monet2155** [comment]
Hello. (I am not good at English.)

I have the same error.

I modified 1149 lines in the framework.js file in the build file, so this error does not occur.

It seems to be a temporary settlement. Please recommend a better way.

```
function _JS_DOM_MapViewportCoordinateToElementLocalCoordinate(
  viewportX,
  viewportY,
  targetX,
  targetY
) {
  var canvasId = Module["canvas"] ? Module["canvas"].id : "unity-canvas";
  var canvasSelector = "#" + jsDomCssEscapeId(canvasId);
  var canvas = document.querySelector(canvasSelector);

// add here
  if(canvas){
      var rect = canvas.getBoundingClientRect();
      HEAPU32[targetX >> 2] = viewportX - rect.left;
      HEAPU32[targetY >> 2] = viewportY - rect.top;
  }
}
```

**Reply by pikktorr** [reply]
@monet2155 Same here. After every webgl build, I have to manually write if(canvas){ ... } in framework.js file. 
It's a shitty solution, but hey it's working.

**Reply by jeffreylanters** [reply]
Hi. Sadly this is an issue related to Unity. The workaround you are mentioning above should prevent the framework from throwing an error. We have to wait for Unity to patch this bug, since their builds are not compatible with unmounting before unloading.

This is a duplicate of #22
